### PR TITLE
Internal: Usage data calculations only for users that opted in. [ED-5070]

### PR DESCRIPTION
**Pull request by iNewLegend**
image

### CODE REVIEW

File: modules/usage/usage-reporter.php [hunk 1]
```diff
@@ -43,6 +44,10 @@ public function get_fields() {
 		];
 	}
 
+	public function is_enabled() {
+		return Tracker::is_allow_track();
```
Comment (matipojo) [inline modules/usage/usage-reporter.php:48]: Please remove also from the post saving

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/upgrade/upgrades.php [hunk 2]
```diff
@@ -640,6 +641,10 @@ public static function _v_2_7_1_remove_old_usage_data() {
 	 * @return bool
 	 */
 	public static function recalc_usage_data( $updater ) {
+		if ( Tracker::is_allow_track() ) {
```
Comment (nisan250) [inline core/upgrade/upgrades.php:644]: If not

**Review by nisan250 — CHANGES_REQUESTED** [review]

File: modules/usage/usage-reporter.php (hunk 1 above)
Comment (iNewLegend) [reply modules/usage/usage-reporter.php]: Handled.

File: core/upgrade/upgrades.php (hunk 2 above)
Comment (iNewLegend) [reply core/upgrade/upgrades.php]: Handled.

**Review by nisan250 — APPROVED** [review]

File: core/upgrade/upgrades.php [hunk 3]
```diff
@@ -641,7 +641,7 @@ public static function _v_2_7_1_remove_old_usage_data() {
 	 * @return bool
 	 */
 	public static function recalc_usage_data( $updater ) {
-		if ( Tracker::is_allow_track() ) {
+		if ( ! Tracker::is_allow_track() ) {
```
Comment (matipojo) [inline core/upgrade/upgrades.php:644]: On optin the usage data - please recalc the usage in the background

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — COMMENTED** [review]
Nice Work! 🔥

**Review by matipojo — APPROVED** [review]

**Pull request merged**